### PR TITLE
Warn when RHS of a claim is \\bottom.

### LINKED ISSUE #2703: Add log and document prover behavior when claim has `\\bottom` RHS
Hello! This is probably an edge case but I think kprove is supposed to validate the `A-SPEC` spec below:
```
module A

syntax Foo ::= "a"

rule a => a

endmodule
```

```
module A-SPEC

imports A

claim a => . ensures false

endmodule
```

But instead it gets stuck at
```
kore-exec: [165966] Warning (WarnStuckClaimState):
    (InfoReachability) while checking the implication:
    The proof has reached the final configuration, but the claimed implication is not valid. Location: /home/rodlin/Desktop/illinois/research/matching-logic-prover/matching-logic-proof-checker/examples/imp/.kprove-2021-06-26-19-29-45-382-5bb92c7f-1d1b-4273-a67b-df0c336a1d60/././a-spec.k:5:7-5:27
<k>
  a ~> _DotVar1 ~> .
</k>
```

I think perhaps it applied the semantics rule first before applying the claim inductively?
The java backend is able to prove this.

### PR DESCRIPTION
**Pull request by RaoulSchaffranek**
Fixes #2703.

---

###### Review checklist

The author performs the actions on the checklist. The reviewer evaluates the work and checks the boxes as they are completed.

-   [ ] **Summary.** Write a summary of the changes. Explain what you did to fix the issue, and why you did it. Present the changes in a logical order. Instead of writing a summary in the pull request, you may push a clean Git history.
-   [ ] **Documentation.** Write documentation for new functions. Update documentation for functions that changed, or complete documentation where it is missing.
-   [ ] **Tests.** Write unit tests for every change. Write the unit tests that were missing before the changes. Include any examples from the reported issue as integration tests.
-   [ ] **Clean up.** The changes are already clean. Clean up anything near the changes that you noticed while working. This does not mean only spatially near the changes, but logically near: any code that interacts with the changes!

### CODE REVIEW

File: kore/src/Kore/Log/WarnClaimRHSIsBottom.hs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+instance Pretty WarnClaimRHSIsBottom where
+    pretty (WarnClaimRHSIsBottom rule) =
+        Pretty.hsep
+            [ "The right-hand side of the claim is bottom:"
+            , Pretty.pretty (from rule :: SourceLocation)
+            ]
+
+instance Entry WarnClaimRHSIsBottom where
+    entrySeverity _ = Warning
+    helpDoc _ = "warn when the right-hand side of a claim is bottom"
```
Comment (JKTKops) [inline kore/src/Kore/Log/WarnClaimRHSIsBottom.hs:31]: I think we should be giving `oneLineDoc` implementations to all new log entries. That feature probably post-dates the entry that you used as a template for this one. The `oneLineDoc` should just be the prettified source location with no other text. See for example `DebugBeginClaim`.

Comment (RaoulSchaffranek) [reply kore/src/Kore/Log/WarnClaimRHSIsBottom.hs:31]: I added the function, but I don't know how to test it. I thought I had to pass `--log-format oneline` to `kprove`, but `kprove` doesn't seem to forward it to `kore-exec`.

Comment (ana-pantilie) [reply kore/src/Kore/Log/WarnClaimRHSIsBottom.hs:31]: If you run `kprove` with the following option `--haskell-backend-command "/path/to/kore-exec --log-format online"` it should work.

File: kore/src/Kore/Reachability/Claim.hs [hunk 2]
```diff
@@ -569,17 +570,19 @@ checkImplicationWorker (ClaimPattern.refreshExistentials -> claimPattern) =
...
+            pure Implied
         | not didAnyUnify
           , not (isBottom right) =
-            NotImplied claimPattern
-        | otherwise =
-            Lens.set (field @"left") stuck claimPattern
-                & NotImpliedStuck
+            pure $ NotImplied claimPattern
+        | otherwise = do
+            warnClaimRHSIsBottom claimPattern
```
Comment (ana-pantilie) [inline kore/src/Kore/Reachability/Claim.hs:582]: We should check if the RHS is bottom before emitting the warning.

**Review by ana-pantilie — CHANGES_REQUESTED** [review]

File: kore/src/Kore/Reachability/Claim.hs (hunk 2 above)
Comment (RaoulSchaffranek) [reply kore/src/Kore/Reachability/Claim.hs]: Obviously, I checked the guards of the preceding cases and falsely concluded that the RHS must always be bottom in the last case. I added the check in 912d7a7

File: kore/src/Kore/Log/WarnClaimRHSIsBottom.hs (hunk 1 above)
Comment (RaoulSchaffranek) [reply kore/src/Kore/Log/WarnClaimRHSIsBottom.hs:31]: Thanks, this worked! Would it be a good idea to add implementations of `oneLineDoc` for all existing entries in this PR, or should I open a fresh issue about that?

Comment (ana-pantilie) [reply kore/src/Kore/Log/WarnClaimRHSIsBottom.hs:31]: I think it's better if we do that separately, so please open an issue. Thanks!

**Review by ana-pantilie — APPROVED** [review]
Looks good! The performance data seems off, I think we need to merge master to get accurate results.

**Review by JKTKops — APPROVED** [review]

**Pull request merged**